src/internal/testutil/auth.go

```diff
@@ -144,16 +144,16 @@ func CommitCnt(t *testing.T, c *client.APIClient, repo *pfs.Repo) int {
 }
 
 // PipelineNames returns the names of all pipelines that 'c' gets from
-// ListPipeline.
-//
-// TODO(CORE-1100): replace with version which knows about projects.
-func PipelineNames(t *testing.T, c *client.APIClient) []string {
+// ListPipeline in the specified project. "" project name means get all pipelines.
+func PipelineNames(t *testing.T, c *client.APIClient, project string) []string {
 	t.Helper()
 	ps, err := c.ListPipeline(false)
 	require.NoError(t, err)
 	result := make([]string, len(ps))
 	for i, p := range ps {
-		result[i] = p.Pipeline.Name
+		if project == "" || project == p.Pipeline.Project.Name {
```

Comment:
Where are we documenting that `""` means "all projects"?

I recommend being paranoid when accessing nested fields in protocol buffers messages; p.GetPipeline().GetProject().GetName().  This returns `""` if any of p, pipeline, or project are `nil`.   The former panics.

Reply:
The function comment mentioned it, but initially I thought I was modelling this off how ListJobs was working. Looking again I misunderstood. I've changed it so that  "" means default since that seems to be the standard behavior elsewhere.